packages/editor/src/components/post-preview-button/index.js

```diff
@@ -210,17 +210,23 @@ export default compose( [
 			getEditedPostPreviewLink,
 		} = select( 'core/editor' );
 		const {
+			getAutosave,
 			getPostType,
+			hasFetchedAutosave,
 		} = select( 'core' );
 
 		const previewLink = getEditedPostPreviewLink();
-		const postType = getPostType( getEditedPostAttribute( 'type' ) );
+		const postTypeName = getEditedPostAttribute( 'type' );
+		const postType = getPostType( postTypeName );
+		const postId = getCurrentPostId();
+		const autosave = getAutosave( postTypeName, postId );
+
 		return {
 			postId: getCurrentPostId(),
 			currentPostLink: getCurrentPostAttribute( 'link' ),
 			previewLink: forcePreviewLink !== undefined ? forcePreviewLink : previewLink,
-			isSaveable: isEditedPostSaveable(),
-			isAutosaveable: forceIsAutosaveable || isEditedPostAutosaveable(),
+			isSaveable: isEditedPostSaveable() && hasFetchedAutosave( postTypeName, postId ),
```

Comment:
nit: `isEnabled` would be a more accurate name for this prop now.

Reply:
Hmm, ok. Would this require a deprecation for the inner component? I'm not sure if the unenhanced `PostPreviewButton` is considered public.

Reply:
> I'm not sure if the unenhanced PostPreviewButton is considered public.

I'd say no: only the exported enhanced component should be considered our public interface.